lib/bamboo/adapters/local_adapter.ex

```diff
@@ -28,9 +28,22 @@ defmodule Bamboo.LocalAdapter do
     SentEmail.push(email)
   end
 
-  @doc "Adds email to Bamboo.SentEmail. Returns a Task that can be awaited on"
-  def deliver_later(email, _config) do
-    deliver(email, nil)
-    Task.async(fn -> :ok end)
+  def handle_config(config) do
+    case config[:deliver_later_strategy] do
+      nil ->
+        Map.put(config, :deliver_later_strategy, Bamboo.DeliverImmediatelyStrategy)
+      Bamboo.DeliverImmediatelyStrategy ->
+        config
+      _ ->
+        raise ArgumentError, """
```

Comment:
You’re right. It probably makes sense to allow any strategy. My one concern was that if someone used some kind of background job library, and something went wrong, or the jobs weren’t run, then the email wouldn’t show up in Bamboo.SentEmail. Now that I think about it though, that’s probably good. If you use some other kind of delivery strategy and it’s not working, you probably want to know about it.

So I think the LocalAdapter should allow any deliver_later_strategy, but what should the default be? Maybe it can be TaskSupervisorStrategy just like everything else. Then the adapter wouldn’t need anything special in handle_config at all.

> On Feb 17, 2016, at 9:22 AM, Steve Domin <notifications@github.com> wrote:
> 
> In lib/bamboo/adapters/local_adapter.ex <https://github.com/paulcsmith/bamboo/pull/71#discussion_r53169414>:
> 
> > @@ -28,9 +28,22 @@ defmodule Bamboo.LocalAdapter do
> >      SentEmail.push(email)
> >    end
> >  
> > -  @doc "Adds email to Bamboo.SentEmail. Returns a Task that can be awaited on"
> > -  def deliver_later(email, _config) do
> > -    deliver(email, nil)
> > -    Task.async(fn -> :ok end)
> > +  def handle_config(config) do
> > +    case config[:deliver_later_strategy] do
> > +      nil ->
> > +        Map.put(config, :deliver_later_strategy, Bamboo.DeliverImmediatelyStrategy)
> > +      Bamboo.DeliverImmediatelyStrategy ->
> > +        config
> > +      _ ->
> > +        raise ArgumentError, """
> Why does that have to be the case? You should be able to use a deliver later strategy even with the local adapter, no?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/paulcsmith/bamboo/pull/71/files#r53169414>.
>